Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskConfig.kt

```diff
@@ -0,0 +1,11 @@
+package de.rki.coronawarnapp.task
+
+interface TaskConfig {
+
+    val executionMode: ExecutionMode
+
+    enum class ExecutionMode {
```

Comment:
Does an execution mode make sense here? When I have two tasks and I want pass them to the controller(?) to execute them, isn't that when I want to specify whether these two tasks should be executed in parallel or sequentially? The task itself shouldn't know/care about this, IMO.

Reply:
So the question would be whether the execution mode is something inherent to the task type?
Should the caller have control over the execution mode, or does the task specify which execution mode it requires?

Maybe the task doesn't support concurrency, then someone calling it with `Mode.Parallel` could break things :thinking: 

If the caller doesn't know about other tasks being started by other callers (possibly of the same type), can it even make the correct decision to pick an execution mode?

Reply:
> So the question would be whether the execution mode is something inherent to the task type?

Exactly.

How about a combination of `supportedExecutionMode` which is inherent to the task and the actual `executionMode`  that's set by the caller?

But that already seems to complex and might seem like we are designing something though we don't know if we'll ever need it.

Anyway, I think having it on task-level only is not correct. What happens if you schedule two tasks, one with ENQUEUE and one with PARALLEL? How would they be executed?

Reply:
> Anyway, I think having it on task-level only is not correct. What happens if you schedule two tasks, one with ENQUEUE and one with PARALLEL? How would they be executed?

Execution mode would be specific to the task type. "TaskType=A (ENQUEUE)" does not intefere with "TaskType=B (PARALLEL)".

Reply:
Do you already have specific scenarios in mind where we _need_ to distinguish between parallel and enqueue? As mentioned above, I strongly recommend not design (for) something where we don't even know if we need it, just because we can 😉

Reply:
I will remove the mode for PARALLEL, the current interfaces allow us to easily readd if necessary.

We need to add a new one though right? `SKIP` The `RetrieveDiagnosisKeysTransaction` transaction currently skips if already running :thinking:

Reply:
We would we even submit a task if it's going to be skipped anyway? Is it the task itself that would decide if it shall be skipped?
Additionally, if we decide to add this to the TaskConfig, I'm wondering whether executionMode is the right place or whether this should be something separate.